cas-server-webapp-actions/src/main/java/org/jasig/cas/web/flow/InitialFlowSetupAction.java

```diff
@@ -67,13 +65,23 @@
     @Override
     protected Event doExecute(final RequestContext context) throws Exception {
         final HttpServletRequest request = WebUtils.getHttpServletRequest(context);
-        if (!this.pathPopulated) {
-            final String contextPath = context.getExternalContext().getContextPath();
-            final String cookiePath = StringUtils.hasText(contextPath) ? contextPath + '/' : "/";
-            logger.info("Setting path for cookies to: {} ", cookiePath);
+
+        final String contextPath = context.getExternalContext().getContextPath();
+        final String cookiePath = StringUtils.isNotBlank(contextPath) ? contextPath + '/' : "/";
+
+        if (StringUtils.isBlank(warnCookieGenerator.getCookiePath())) {
```

Comment:
what I meant was `StringUtils.isBlank(warnCookieGenerator.getCookiePath())` will always return false because the `cookiePath` in the `CookieGenerator` defaults to "/" and not null/an empty string.

So, the autogeneration logic of the `cookiePath` value won't run unless we specifically initialize the `CookieGenerator`s to a blank value.